onnxscript/type_annotation.py

```diff
@@ -30,31 +30,91 @@
 _LIST_CONSTRUCTORS = frozenset([list, typing.List, typing.Sequence, collections.abc.Sequence])
 
 
-def pytype_to_attrtype(pytype: type) -> typing.Optional[onnx.AttributeProto.AttributeType]:
+def remove_annotation(typeinfo):
+    """Remove Annotated wrapper if present, otherwise return typeinfo as is."""
+    if hasattr(typing, "Annotated"):
+        # Present in Python 3.9+
+        if get_origin(typeinfo) is typing.Annotated:
+            return get_args(typeinfo)[0]
+    return typeinfo
+
+
+def is_primitive_attr_type(typeinfo) -> bool:
+    return typeinfo in _PYTYPE_TO_ATTRTYPE_MAP
+
+
+def pytype_to_attrtype(pytype) -> typing.Optional[onnx.AttributeProto.AttributeType]:
+    pytype = remove_annotation(pytype)
     if pytype in _PYTYPE_TO_ATTRTYPE_MAP:
         return _PYTYPE_TO_ATTRTYPE_MAP[pytype]
-    if get_origin(pytype) in _LIST_CONSTRUCTORS:
-        args = get_args(pytype)
-        elt_type = args[0]
+    type_constructor = get_origin(pytype)
+    # Remove Optional wrapper if present, which is represented as an Union[..., None]
+    if type_constructor is typing.Union:
+        # Filter out None, since typing.Optional[X] evaluates to Union[X, None]
+        args = [x for x in get_args(pytype) if x is not type(None)]  # noqa: E721
+        if len(args) == 1:
+            return pytype_to_attrtype(args[0])
```

Comment:
Is it true that we still use None when there are multiple types? Can the info be stored somewhere?